src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -95,7 +98,7 @@ public String getScript() {
             String defaultShell = jenkins.getInjector().getInstance(Shell.DescriptorImpl.class).getShellOrDefault(ws.getChannel());
             s = "#!"+defaultShell+" -xe\n" + s;
         }
-        shf.write(s, "UTF-8");
+        shf.write(s, encoding==null ? "UTF-8" : encoding);
```

Comment:
Would we not just want to use the system encoding on the agent, regardless of OS?